docs/userguide/tasks.rst

```diff
@@ -2054,7 +2054,31 @@ once all transactions have been committed successfully.
     @transaction.atomic
     def create_article(request):
         article = Article.objects.create()
-        transaction.on_commit(lambda: expand_abbreviations.delay(article.pk))
+        expand_abbreviations.delay_on_commit(article.pk)
+        return HttpResponseRedirect('/articles/')
+
+This method was added in Celery 5.4. It's a shortcut that uses Django's
+``on_commit`` callback to launch your Celery task once all transactions
+have been committed successfully.
+
+With Celery <5.4
+~~~~~~~~~~~~~~~~
```

Comment:
Left the old instructions for earlier versions of Celery, but perhaps we can entirely remove the old way since the docs for the older versions are accessible as e.g. https://docs.celeryq.dev/en/v5.3.6/userguide/tasks.html

Reply:
It’s not bad to say that in an older version it behaves differently, but I agree with your POV/logic.